### PR TITLE
Setup unit testing with karma (fixes #39)

### LINKED ISSUE #39: Setup unit testing with karma + webpack

### PR DESCRIPTION
**Pull request by muffinresearch**
Adds a single test running under the karma-webpack setup.

### CODE REVIEW

File: config/karma.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+var grunt = require('grunt');
+var testOption = grunt.option('tests');
```
Comment (kumar303) [inline config/karma.js]: this doesn't look it's used. Were you adding a way to run a single file? That would be nice. The way it works for fxpay is pretty useful: https://github.com/mozilla/fxpay#running-tests

File: config/karma.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+module.exports  = {
+  options: {
+    logLevel: grunt.option('log-level') || 'ERROR',
+  },
+  dev: {
+    configFile: 'karma.conf.js',
+    autoWatch: true
```
Comment (kumar303) [inline config/karma.js]: I found this pretty buggy but I guess we could try it again

File: public/js/main.js [hunk 3]
```diff
@@ -4,30 +4,6 @@ var $ = require('jquery');
 var braintree = require('braintree-web');
 
 // The core field props passed into <CardDetails>
```
Comment (kumar303) [inline public/js/main.js]: does this comment still apply?

File: tests/test.index.js [hunk 4]
```diff
@@ -0,0 +1,2 @@
+var context = require.context('.', true, /test\..*?.jsx$/);
+context.keys().forEach(context);
```
Comment (kumar303) [inline tests/test.index.js]: what does this do?

**Comment by kumar303** [comment]
Nice start!

This might be a good time to drop a note in the readme for how to run the tests.

r+wc

File: config/karma.js (hunk 1 above)
Comment (muffinresearch) [inline config/karma.js]: I was going to. The recommended way that the webpack tests are setup means the tests are bundled. This is the fastest way to have tests running. I'm going to start with that and come back to dealing with the single test case if it comes up later.

File: tests/test.index.js (hunk 4 above)
Comment (muffinresearch) [inline tests/test.index.js]: I've added a comment.

**Comment by muffinresearch** [comment]
> This might be a good time to drop a note in the readme for how to run the tests.

Done, thanks for catching all the nits!

File: tests/test.index.js (hunk 4 above)
Comment (kumar303) [inline tests/test.index.js]: aha, nifty

**Pull request merged**